### PR TITLE
Add typescript definition

### PR DESCRIPTION
**Pull request by jean9696**
Thanks for this cool library! We use it really often and not having a typescript definition was a shame so I created it. Hope it will suit you.
I based my work on the [jsdoc](https://www.marzipano.net/reference/index.html)

Relates to https://github.com/google/marzipano/issues/329

### CODE REVIEW

**Comment by jean9696** [comment]
@googlebot I signed it!

File: src/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,1194 @@
...
+    constructor(parameter: string)
+
+    destroy(): boolean
+
+    setFriction(friction: number): void
+
+    setVelocity(velocity: number): void
+  }
+
+  export interface VelocityControlMethodInterface
```
Comment (JoeScho) [inline src/types.d.ts:749]: This is duplicated

File: src/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,1194 @@
...
+
+    startMovement(fn: Function, done?: Function): void
+
+    stopMovement(): void
+
+    switchTo(opts?: SwitchSceneOptions, done?: Function): void
+
+    view(): View
+
+    viewer(): Viewer
```
Comment (JoeScho) [inline src/types.d.ts:625]: this is duplicated

File: src/types.d.ts [hunk 3]
```diff
@@ -0,0 +1,1194 @@
...
+    pitch(min: number, max: number): RectilinearViewLimiter
+
+    resolution(size: number): RectilinearViewLimiter
+
+    roll(min: number, max: number): RectilinearViewLimiter
+
+    traditional(
+      maxResolution: number,
+      maxVFov: number,
+      maxHFox
```
Comment (pmuller) [inline src/types.d.ts:542]:
```suggestion
      maxHFov?: number
```

File: src/types.d.ts (hunk 1 above)
Comment (pmuller) [reply src/types.d.ts:749]: What's the purpose of the VelocityControlMethodInterface? When are we supposed to use it?

File: src/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,1194 @@
...
+
+    disableMethodGroup(groupId: string): void
+
+    enable(): void
+
+    enabled(): boolean
+
+    enableMethod(id: string): void
+
+    enableMethodGroup(groupId): void
```
Comment (pmuller) [inline src/types.d.ts:64]:
```suggestion
    enableMethodGroup(groupId: string): void
```

File: src/types.d.ts [hunk 5]
```diff
@@ -0,0 +1,1194 @@
...
+    constructor(levelPropertiesList: { size: number; tileSize: number }[])
+  }
+
+  /**
+   * CubeTile
+   */
+  export class CubeTile implements Tile {
+    cmp(that): number
+
+    equals(that): boolean
```
Comment (pmuller) [inline src/types.d.ts:92]:
```suggestion
    cmp(that: Tile): number

    equals(that: Tile): boolean
```

File: src/types.d.ts [hunk 6]
```diff
@@ -0,0 +1,1194 @@
...
+      parameter: string,
+      velocity: number,
+      friction: number
+    )
+
+    destroy(): void
+  }
+
+  export interface ElementPressControlMethodInterface
+    extends ElementPressControlMethod {}
```
Comment (pmuller) [inline src/types.d.ts:137]: What's the purpose of this interface?

File: src/types.d.ts [hunk 7]
```diff
@@ -0,0 +1,1194 @@
...
+    type: 'equirect'
+  }
+
+  /**
+   * EquirectTile
+   */
+  export class EquirectTile implements Tile {
+    cmp(that): number
+
+    equals(that): boolean
```
Comment (pmuller) [inline src/types.d.ts:154]:
```suggestion
    cmp(that: Tile): number

    equals(that: Tile): boolean
```

File: src/types.d.ts [hunk 8]
```diff
@@ -0,0 +1,1194 @@
...
+    type: 'flat'
+  }
+
+  /**
+   * FlatTile
+   */
+  export abstract class FlatTile implements Tile {
+    cmp(that): number
+
+    equals(that): boolean
```
Comment (pmuller) [inline src/types.d.ts:181]:
```suggestion
    cmp(that: Tile): number

    equals(that: Tile): boolean
```

File: src/types.d.ts [hunk 9]
```diff
@@ -0,0 +1,1194 @@
...
+        cubeMapPreviewUrl?: string
+        retryDelay?: number
+      }
+    ) => ImageUrlSource
+
+    loadAsset(stage: Stage, tile: Tile, done: Function): Function
+
+    addEventListener(event: Event, callback: Function)
+
+    removeEventListener(event: Event, callback: Function)
```
Comment (pmuller) [inline src/types.d.ts:363]: `Function` is rather vague.

We should define custom types for those callbacks. However, those are not documented and I am not sure I get them right when reading the JS code...

File: src/types.d.ts [hunk 10]
```diff
@@ -0,0 +1,1194 @@
...
+  /**
+   * NetworkError
+   */
+  export class NetworkError {}
+
+  /**
+   * PinchZoomControlMethod
+   */
+  export class PinchZoomControlMethod {
+    constructor(element: HTMLElement, pointerType: string, opts: any)
```
Comment (pmuller) [inline src/types.d.ts:439]: The [docs](https://www.marzipano.net/reference/PinchZoomControlMethod.html) are not helpful regarding `opts`.

According to the [code](https://github.com/google/marzipano/blob/eb116412d1b7c6df08906e9affcc9d3d65b5906f/src/controls/PinchZoom.js#L34), this parameter is not even in use.

Maybe we should not expose it in the type definitions.

File: src/types.d.ts [hunk 11]
```diff
@@ -0,0 +1,1194 @@
...
+
+    size(size?: Size): Size
+
+    setSize(size: Size): void
+
+    width(): number
+
+    type: 'rectilinear'
+
+    coordinatesToPerspectiveTransform(coords, radius, extraTransforms): string
```
Comment (pmuller) [inline src/types.d.ts:487]:
```suggestion
    coordinatesToPerspectiveTransform(
      coords: RectilinearViewCoords,
      radius: number,
      extraTransforms: string
    ): string
```

File: src/types.d.ts [hunk 12]
```diff
@@ -0,0 +1,1194 @@
...
+  }
+
+  export interface HotspotOptions {
+    perspective?: Perspective
+  }
+
+  /**
+   * HotspotContainer
+   */
+  export class HotpsotContainer {
```
Comment (pmuller) [inline src/types.d.ts:296]:
```suggestion
  export class HotspotContainer {
```

File: src/types.d.ts [hunk 13]
```diff
@@ -0,0 +1,1194 @@
...
+      layerOpts?: LayerOptions
+    }): Layer
+
+    destroy(): void
+
+    destroyAllLayers(): void
+
+    destroyLayer(layer: Layer): void
+
+    hotspotContainer(): HotpsotContainer
```
Comment (pmuller) [inline src/types.d.ts:605]:
```suggestion
    hotspotContainer(): HotspotContainer
```

**Pull request closed without merging**